flow-server/src/main/java/com/vaadin/flow/component/page/WebStorage.java

```diff
@@ -215,17 +219,88 @@ public static void getItem(Storage storage, String key, Callback callback) {
      *            the key for which the value will be fetched
      * @param callback
      *            the callback that gets the value once transferred from the
-     *            client side
+     *            client side or <code>null</code> if the value was not set
      */
     public static void getItem(UI ui, Storage storage, String key,
             Callback callback) {
-        ui.getPage()
-                .executeJs("return window[$0].getItem($1);", storage.toString(),
-                        key)
-                .then(String.class, callback::onValueDetected, s -> {
-                    // for error (most likely non-existing mapping), return null
+        requestItem(ui, storage, key).then(String.class,
+                callback::onValueDetected, s -> {
+                    LoggerFactory.getLogger(WebStorage.class.getName()).debug(
+                            "Error while getting value for key '{}' from storage '{}': {}",
+                            key, storage, s);
+                    // fallback to null if there was an error
                     callback.onValueDetected(null);
                 });
     }
 
+    /**
+     * Asynchronously gets an item from the local storage.
+     * <p>
+     * It is not possible to synchronously wait for the result of the execution
+     * while holding the session lock since the request handling thread that
+     * makes the result available will also need to lock the session. <br>
+     * See {@link PendingJavaScriptResult#toCompletableFuture} for more
+     * information.
+     *
+     * @param key
+     *            the key for which the value will be fetched
+     * @return a CompletableFuture that will be completed with the value once
+     *         transferred from the client side or <code>null</code> if the
+     *         value was not set
```

Comment:
I would use `available` as in the `Callback` javadoc
```suggestion
     *         value was not available.
```